### PR TITLE
[String] Add singularize and pluralize filter

### PR DESCRIPTION
**Pull request by welcoMattic**
This PR follows up this one in Symfony https://github.com/symfony/symfony/pull/57245.

It adds 2 new filters `singularize` and `pluralize`. It uses the String component, especially `EnglishInflector` and `FrenchInflector`.

Examples:

```php
{{ 'person'|pluralize('en') }} // persons
{{ 'person'|pluralize('en', false) }} // ['persons', 'people']


{{ 'personne'|pluralize('fr') }} // personnes
{{ 'personne'|pluralize('fr', false) }} // ['personnes']


{{ 'persons'|singluarize('en') }} // person
{{ 'people'|singluarize('en') }} // person
{{ 'persons'|singluarize('en', false) }} // ['person']

{{ 'personnes'|singluarize('fr') }} // personne
{{ 'personnes'|singluarize('fr', false) }} // ['personne']



{{ 'persona'|pluralize('it') }} // \InvalidArgumentException('Language "it" is not supported.')
{{ 'persone'|singluarize('it') }} // \InvalidArgumentException('Language "it" is not supported.')
```

### CODE REVIEW

File: extra/string-extra/StringExtension.php [hunk 1]
```diff
@@ -44,4 +51,42 @@ public function createSlug(string $string, string $separator = '-', ?string $loc
     {
         return $this->slugger->slug($string, $separator, $locale);
     }
+
+    /**
+     * @return array|string
+     */
+    public function pluralize($value, string $lang, bool $singleResult = true)
```
Comment (smnandre) [inline extra/string-extra/StringExtension.php]:
```suggestion
    public function pluralize(string $value, string $lang, bool $singleResult = true)
```

File: extra/string-extra/StringExtension.php [hunk 2]
```diff
@@ -44,4 +51,42 @@ public function createSlug(string $string, string $separator = '-', ?string $loc
...
+                return $this->getInflector($lang)->pluralize($value)[0];
+            default:
+                return $this->getInflector($lang)->pluralize($value);
+        }
+    }
+
+    /**
+     * @return array|string
+     */
+    public function singularize($value, string $lang, bool $singleResult = true)
```
Comment (smnandre) [inline extra/string-extra/StringExtension.php]:
```suggestion
    public function singularize(string $value, string $lang, bool $singleResult = true)
```

File: extra/string-extra/README.md [hunk 3]
```diff
@@ -10,7 +10,13 @@ object to give access to [methods of the class][2].
 It also provides a [`slug`][3] filter which is simply a wrapper for the 
 [`AsciiSlugger`][4]'s `slug` method.
 
+In addition, two filters are provided to [`singularize`][5] and [`pluralize`][6] a string.
```
Comment (smnandre) [inline extra/string-extra/README.md]:
```suggestion
In addition, two filters are provided to [`singularize`][5] and [`pluralize`][6] a noun. 
```

File: doc/filters/singularize.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        use Twig\Extra\String\StringExtension;
+
+        $twig = new \Twig\Environment(...);
+        $twig->addExtension(new StringExtension());
+
+Arguments
+---------
+
+* ``lang``: The lang of the original string. Only English (`en`) and French (`fr`) are supported.
+* ``singleResult``: This argument is optional. If set to false, the filter will return an array of singularized words. Default is true.
```
Comment (smnandre) [inline doc/filters/singularize.rst]:
```suggestion
* ``singleResult``: This argument is optional. If set to false, the filter will return an array of singularized nouns. Default is true.
```

File: doc/filters/singularize.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
+``singularize``
+========
+
+The ``singularize`` filter transforms a given string into its singular version.
```
Comment (smnandre) [inline doc/filters/singularize.rst]:
```suggestion
The ``singularize`` filter transforms a given noun in its plural form into its singular version.
```

File: doc/filters/pluralize.rst [hunk 6]
```diff
@@ -0,0 +1,45 @@
+``pluralize``
+========
+
+The ``pluralize`` filter transforms a given string into its plural version.
```
Comment (smnandre) [inline doc/filters/pluralize.rst]:
```suggestion
The ``pluralize`` filter transforms a given noun in its singular form into its plural version.
```

**Review by smnandre — APPROVED** [review]

**Comment by fabpot** [comment]
I've just pushed an additional commit that makes some significant changes. It contains many tweaks for the docs and the code, but most importantly, it renames the filters to `singular` and `plural`. Tell me what you think about these changes.

File: extra/string-extra/Tests/Fixtures/plural-invalid-language.test [hunk 7]
```diff
@@ -0,0 +1,10 @@
+--TEST--
+"plural" filter
+--TEMPLATE--
+{{ 'partition'|plural('it') }}
```
Comment (GromNaN) [inline extra/string-extra/Tests/Fixtures/plural-invalid-language.test:4]: If that can motivate someone to implement pluralization in Italian, just to break this test 😂

**Comment by fabpot** [comment]
Thank you @welcoMattic.

**Pull request merged**